2.19.x typing changes break compatibility with openlayers

Heya! Thanks for this project :)

Our project [needed to upper bound proj4js <2.19.0](https://github.com/geojupyter/jupytergis/pull/778) after the recent release because our project fails to build. I created a minimal reproducible example: https://github.com/mfisher87/sscce-proj4js-2.19.x-incompat-with-ol

You should be able to repro following the instructions in the README.

```
$ npm run build

> ol-proj4-compat@0.0.1 build
> tsc -p .

index.ts:4:10 - error TS2345: Argument of type 'ProjectionDefinition' is not assignable to parameter of type 'typeof import("/home/robatt/_test/proj4-type/ol-proj4-compat/node_modules/proj4/dist/lib/index")'.
  Property 'proj4' is missing in type 'ProjectionDefinition' but required in type 'typeof import("/home/robatt/_test/proj4-type/ol-proj4-compat/node_modules/proj4/dist/lib/index")'.

4 register(proj4.defs("EPSG:3413"));
           ~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/proj4/dist/lib/index.d.ts:1:1
    1 export default proj4;
      ~~~~~~~~~~~~~~~~~~~~~
    'proj4' is declared here.


Found 1 error in index.ts:4
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,0 +1,3 @@
+import proj4 from "./lib/index";
+export = proj4;
+export as namespace proj4;

--- a/package.json
+++ b/package.json
@@ -5,10 +5,10 @@
   "homepage": "https://proj4js.github.io/proj4js/",
   "main": "dist/proj4-src.js",
   "module": "lib/index.js",
-  "types": "dist/lib/index.js",
+  "types": "dist/index.d.ts",
   "scripts": {
     "prepare": "npm run test",
-    "build": "grunt && npm run rollup",
+    "build": "grunt && npm run rollup && shx cp index.d.ts dist/",
     "build:tmerc": "grunt build:tmerc && npm run rollup",
     "rollup": "rollup -c",
     "pretest": "npm run lint && npm run types",
@@ -42,6 +42,7 @@
     "nyc": "^17.1.0",
     "puppeteer": "^24.2.0",
     "rollup": "^4.9.6",
+    "shx": "^0.4.0",
     "typescript": "^5.8.3"
   },
   "dependencies": {

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -8,7 +8,6 @@
     "declarationDir": "dist",
   },
   "include": [
-    "lib/index.d.ts",
     "lib/**/*.js",
-  ],
+  ]
 }

